### PR TITLE
Convert computational graph diagrams to Mermaid

### LINKED ISSUE #3580: Issue: Incorrect `requires_grad` in the last diagram
In the tutorial [Understanding requires_grad, retain_grad, Leaf, and Non-leaf Tensors](https://docs.pytorch.org/tutorials/beginner/understanding_leaf_vs_nonleaf_tutorial.html), the **last diagram** (showing the computational graph) has an error.

It shows `z = x @ W + b` with `requires_grad = False`.  
However, since `W` and `b` both have `requires_grad = True`, the output `z` should **also** have `requires_grad = True`.

This is misleading for learners, as PyTorch automatically sets `requires_grad = True` for any tensor that depends on a tensor requiring gradients.

Please correct the diagram to reflect this.

Thank you!

Image

### PR DESCRIPTION
**Pull request by saurabhkthakur**
Fixes #3580

## Description
- corrected requires_grad value for z in computational graph
- recreated both computational graph diagrams using Mermaid
forward computation:
image
backward computation:
image

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [ ] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by sekyondaMeta** [comment]
Closing this one in favor of #3873
